raft.c

```diff
@@ -496,12 +496,19 @@ static int raftPersistTerm(raft_server_t *raft, void *user_data, raft_term_t ter
 static int raftApplyLog(raft_server_t *raft, void *user_data, raft_entry_t *entry, raft_index_t entry_idx)
 {
     RedisRaftCtx *rr = user_data;
+    RaftReq *raft_req = entry->user_data;
     RaftCfgChange *req;
 
     switch (entry->type) {
         case RAFT_LOGTYPE_DEMOTE_NODE:
             if (rr->state == REDIS_RAFT_UP && raft_is_leader(rr->raft)) {
                 raft_entry_t *rem_entry = raft_entry_new(sizeof(RaftCfgChange));
+
```

Comment:
Maybe it would be a good idea to skip this if `raft_req` is already NULL, there are different scenarios that could lead to that.

Reply:
as I understand this comment, what you are arguing is that it makes no sense to add a "REMOVE" entry (after demote) on any node besides the leader (where raft_req would exist)?

Reply:
though, my understanding of the comment can't be right, as we already have a check for "raft_is_leader(rr->raft)

Reply:
The leader could have been re-elected in between.

Reply:
ok, now I understand.  can't handle raft_req issues in the if leader because the node with the req might not be the leader anymore.  which also explains why the break was within the if leader as the fall through was by design.